src/proto/h1/role.rs

```diff
@@ -359,6 +361,217 @@ impl Http1Transaction for Server {
             extend(dst, b"\r\n");
         }
 
+        let orig_headers;
+        let extensions = mem::take(&mut msg.head.extensions);
```

Comment:
Instead of `take` here, you could do `msg.head.extensions.remove::<HeaderCaseMap>()`, yea?

Reply:
I didn't want to use `Extensions::remove` because I didn't want to mutate that for no reason given we just need a reference to the header case map, so `take` felt less heavy to me. I can change it anyway.

Reply:
But `take` is mutating the `msg.head` anyways, right?

Reply:
Yeah yeah, I mean doing what amounts to a simple `memcpy` (moving `msg.head.extensions` out) vs mutating a glorified hashmap (calling `Extensions::remove`).

Reply:
I tried just not moving out anything by splitting `msg` into more arguments to `encode_headers`, but that was way too many arguments for me to be confortable committing that.